### PR TITLE
mingw buildability

### PR DESCRIPTION
**Pull request by lyrra**
- [ x] I signed the [CLA](https://musescore.org/en/cla)

### CODE REVIEW

File: build/cmake/FindSndFile.cmake [hunk 1]
```diff
@@ -1,6 +1,6 @@
 include(GetPlatformInfo)
 
-if (OS_IS_WIN)
+if (OS_IS_WIN AND (NOT MINGW))
     find_path(SNDFILE_INCDIR sndfile.h PATHS ${PROJECT_SOURCE_DIR}/dependencies/include;)
     if (MINGW)
```
Comment (Jojo-Schmitz) [inline build/cmake/FindSndFile.cmake:5]: wouldn't this then be dead code?

**Comment by Jojo-Schmitz** [comment]
When I try thi,s I get:
```
-- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE) 
CMake Error at build/cmake/FindSndFile.cmake:108 (message):
  Could not find: sndfile
Call Stack (most recent call first):
  CMakeLists.txt:169 (include)
```

File: build/cmake/FindSndFile.cmake (hunk 1 above)
Comment (lyrra) [reply build/cmake/FindSndFile.cmake:5]: Originally the path was took for all flavors of windows. This narrows to a subset that does not include mingw.

**Comment by lyrra** [comment]
> -- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE) 
> CMake Error at build/cmake/FindSndFile.cmake:108 (message):
>   Could not find: sndfile

Not sure about the PKG_CONFIG_EXECUTABLE error, but check if your mingw has libsndfile is installed:
https://packages.msys2.org/package/mingw-w64-x86_64-libsndfile?repo=mingw64

**Comment by Jojo-Schmitz** [comment]
I'm pretty sure it has, as my 3.x builds find it

File: build/cmake/FindSndFile.cmake (hunk 1 above)
Comment (Jojo-Schmitz) [reply build/cmake/FindSndFile.cmake:5]: But causes sndfile to not get found

**Comment by lyrra** [comment]
It seems pkg-config is needed to be found, and if so, it is used to find libsndfile:
```
-- Found PkgConfig: C:/Users/larry/Downloads/msys64/mingw64/bin/pkg-config.exe (found version "1.8.0")
-- Found sndfile: C:/Users/larry/Downloads/msys64/mingw64/lib/libsndfile.dll.a C:/Users/larry/Downloads/msys64/mingw64/include

```

```
$ command -v pkg-config
/mingw64/bin/pkg-config
```

File: build/cmake/FindSndFile.cmake (hunk 1 above)
Comment (lyrra) [reply build/cmake/FindSndFile.cmake]: I see what you mean with dead code, I've removed the mingw stuff inside the windows path that is avoided on mingw.

**Comment by Jojo-Schmitz** [comment]
I'm using the MinGW that comes as part of Qt 5.15.2, maybe that's just too old?

**Comment by lyrra** [comment]
> I beleive `MUE_BUILD_CRASHPAD_CLIENT` and `MUE_BUILD_MUSESAMPLES_MODULE` need to get disabled, at lerast I can't build with it in QtCreator/MinGW. Also the entire WASAPI stuff doesn't build for me.
> 
> I'm using the MinGW that comes as part of Qt 5.15.2, maybe that's just too old?

The WASAPI errors are expected as this PR depends on #19246 which disables wasapi if jack is used (which is needed for mingw).

Yes, also other stuff needs to be disabled, here is my build command:
```
cmake \
    -P ./build.cmake \
    -G "Unix Makefiles" \
    -DMUE_ENABLE_AUDIO_JACK=ON \
    -DMUE_BUILD_CRASHPAD_CLIENT=OFF \
    -DCMAKE_BUILD_TYPE=Debug \
    -DBUILD_FOR_WINSTORE=OFF \
    -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON
```

**Review by igorkorsukov — APPROVED** [review]

**Comment by Jojo-Schmitz** [comment]
But isn't https://github.com/musescore/MuseScore/pull/19246 just about Linux?
Whatever: even with `MUE_BUILD_CRASHPAD_CLIENT`, `MUE_BUILD_MUSESAMPLES_MODULE` set to `OFF` and `MUE_ENABLE_AUDIO_JACK`tp ON, it doesn't build for me:

```
C:\Users\Jojo\Documents\GitHub\MuseScore\src\framework\audio\internal\platform\win\wasapitypes.h:28: Fehler: winrt/Windows.Devices.Enumeration.h: No such file or directory
In file included from C:/Users/Jojo/Documents/GitHub/MuseScore/src/framework/audio/internal/platform/win/wasapiaudioclient.h:26,
                 from C:/Users/Jojo/Documents/GitHub/MuseScore/src/framework/audio/internal/platform/win/wasapiaudioclient.cpp:23,
                 from C:\Users\Jojo\Documents\GitHub\MuseScore\build.qtc\src\framework\audio\CMakeFiles\audio.dir\Unity\unity_1_cxx.cxx:9:
C:/Users/Jojo/Documents/GitHub/MuseScore/src/framework/audio/internal/platform/win/wasapitypes.h:28:10: fatal error: winrt/Windows.Devices.Enumeration.h: No such file or directory
 #include <winrt/Windows.Devices.Enumeration.h>
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
C:\Users\Jojo\Documents\GitHub\MuseScore\src\framework\audio\audiomodule.cpp:126: Fehler: invalid application of 'sizeof' to incomplete type 'mu::audio::AudioDevicesListener'
In file included from C:/Qt/Tools/mingw810_64/lib/gcc/x86_64-w64-mingw32/8.1.0/include/c++/memory:80,
                 from C:/Users/Jojo/Documents/GitHub/MuseScore/build/pch/pch.h:39,
                 from C:/Users/Jojo/Documents/GitHub/MuseScore/build.qtc/src/framework/global/CMakeFiles/global.dir/cmake_pch.hxx:5,
                 from <command-line>:
C:/Qt/Tools/mingw810_64/lib/gcc/x86_64-w64-mingw32/8.1.0/include/c++/bits/unique_ptr.h: In instantiation of 'void std::default_delete<_Tp>::operator()(_Tp*) const [with _Tp = mu::audio::AudioDevicesListener]':
C:/Qt/Tools/mingw810_64/lib/gcc/x86_64-w64-mingw32/8.1.0/include/c++/bits/unique_ptr.h:274:17:   required from 'std::unique_ptr<_Tp, _Dp>::~unique_ptr() [with _Tp = mu::audio::AudioDevicesListener; _Dp = std::default_delete<mu::audio::AudioDevicesListener>]'
C:/Users/Jojo/Documents/GitHub/MuseScore/src/framework/audio/internal/platform/win/wasapiaudiodriver.h:34:7:   required from 'std::__shared_count<_Lp>::__shared_count(_Ptr) [with _Ptr = mu::audio::WasapiAudioDriver*; __gnu_cxx::_Lock_policy _Lp = (__gnu_cxx::_Lock_policy)2]'
C:/Qt/Tools/mingw810_64/lib/gcc/x86_64-w64-mingw32/8.1.0/include/c++/bits/shared_ptr_base.h:620:22:   required from 'std::__shared_count<_Lp>::__shared_count(_Ptr, std::false_type) [with _Ptr = mu::audio::WasapiAudioDriver*; __gnu_cxx::_Lock_policy _Lp = (__gnu_cxx::_Lock_policy)2; std::false_type = std::integral_constant<bool, false>]'
C:/Qt/Tools/mingw810_64/lib/gcc/x86_64-w64-mingw32/8.1.0/include/c++/bits/shared_ptr_base.h:1107:64:   required from 'std::__shared_ptr<_Tp, _Lp>::__shared_ptr(_Yp*) [with _Yp = mu::audio::WasapiAudioDriver; <template-parameter-2-2> = void; _Tp = mu::audio::IAudioDriver; __gnu_cxx::_Lock_policy _Lp = (__gnu_cxx::_Lock_policy)2]'
C:/Qt/Tools/mingw810_64/lib/gcc/x86_64-w64-mingw32/8.1.0/include/c++/bits/shared_ptr.h:139:46:   required from 'std::shared_ptr<_Tp>::shared_ptr(_Yp*) [with _Yp = mu::audio::WasapiAudioDriver; <template-parameter-2-2> = void; _Tp = mu::audio::IAudioDriver]'
C:/Users/Jojo/Documents/GitHub/MuseScore/src/framework/audio/audiomodule.cpp:126:74:   required from here
C:/Qt/Tools/mingw810_64/lib/gcc/x86_64-w64-mingw32/8.1.0/include/c++/bits/unique_ptr.h:79:16: error: invalid application of 'sizeof' to incomplete type 'mu::audio::AudioDevicesListener'
  static_assert(sizeof(_Tp)>0,
                ^~~~~~~~~~~
```

**Comment by lyrra** [comment]
You are correct, you need this patch to make jack build also on windows: https://github.com/musescore/MuseScore/pull/19246/commits/c5ec4e1cb850d988b61eed0fdcdfd90b006d9017

**Pull request merged**